### PR TITLE
fix: app.relaunch loses args when execPath specified

### LINKED ISSUE #33686: [Bug]: app.relaunch loses args when execPath specified.
### Preflight Checklist

- [X] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/main/CONTRIBUTING.md) for this project.
- [X] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/electron/electron/issues) for a bug report that matches the one I want to file, without success.

### Electron Version

18.0.1

### What operating system are you using?

macOS

### Operating System Version

10.14.6

### What arch are you using?

x64

### Last Known Working Electron version

17.0.0-alpha.3

### Expected Behavior

When invoking `app.relaunch` and specifying both `execPath` and `args`, I expect the args to be passed to the new process.

### Actual Behavior

The binary specified by `execPath` is spawned as a new process, but the `args` are not applied.

### Testcase Gist URL

https://gist.github.com/p120ph37/98cf0fb13790feed702f3e90cbddfbf6

### Additional Information

This bug was introduced in `17.0.0-alpha.4`, and is present in all newer stable versions up to and including at least `19.0.0-alpha.1`  Also note that `app.relaunch()` doesn't work at all from within Electron Fiddle, so to run this gist, you must launch Electron directly (e.g.: via `npm start`).

### PR DESCRIPTION
**Pull request by p120ph37**
I think this should fix #33686, but I didn't have a local toolchain to do the full build/test.  However, CircleCI was nice enough to build this PR for me, and I tested the resulting Electron artifact using my [gist from 33686]( https://gist.github.com/p120ph37/98cf0fb13790feed702f3e90cbddfbf6), and it fixes the issue for me!

#### Description of Change

The change from bitwise to boolean "or"-operator in 1e37162 caused a short-circuit of the evaluation of a function-call with side-effects.  This change pre-evaluates the functions and assigns the results to separate local variables to ensure the side-effects apply in all cases, then combines the variables with the short-circuit-able boolean operator in the conditional.

Fixes #33686

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] tests are [changed or added](https://github.com/electron/electron/blob/main/docs/development/testing.md)
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: Fixed an issue where app.relaunch loses args when execPath is specified.

### CODE REVIEW

**Comment by RaisinTen** [comment]
> I don't have a local toolchain to do the full build/test. Could someone please check this for me?

Maybe you could write a test with [your gist](https://gist.github.com/p120ph37/98cf0fb13790feed702f3e90cbddfbf6) and we'll know if your fix works if it passes on CI?

**Comment by codebytere** [comment]
@p120ph37 please fill out the PR template you removed in order for us to continue reviewing this PR.

**Comment by p120ph37** [comment]
@RaisinTen , I wasn't able to test initially, because I don't have a working build toolchain, but after filing this PR, I was able to use the CircleCI build output to test using my gist, and it solves the issue for me.

@codebytere , now that I've been able to actually test this code-change with the CircleCI output after the initial skeleton PR, I've filled in the full PR template.

**Comment by RaisinTen** [comment]
> I wasn't able to test initially, because I don't have a working build toolchain, but after filing this PR, I was able to use the CircleCI build output to test using my gist, and it solves the issue for me.

That's good to know :+1:. A test would still be beneficial though. It would prevent further regressions in the future.

File: spec-main/api-app-spec.ts [hunk 1]
```diff
@@ -370,9 +370,11 @@ describe('app module', () => {
       server!.once('error', error => done(error));
       server!.on('connection', client => {
         client.once('data', data => {
-          if (String(data) === 'false' && state === 'none') {
+          if (String(data) === '--first' && state === 'none') {
             state = 'first-launch';
-          } else if (String(data) === 'true' && state === 'first-launch') {
+          } else if (String(data) === '--second' && state === 'first-launch') {
+            state = 'second-launch';
+          } else if (String(data) === '--third' && state === 'second-launch') {
```
Comment (RaisinTen) [inline spec-main/api-app-spec.ts:377]: this test seems to be timing out on CI

**Review by RaisinTen — APPROVED** [review]
Thanks, LGTM!

**Review by codebytere — APPROVED** [review]

**Pull request merged**